lib/rules/no-useless-escape.js

```diff
@@ -95,10 +114,12 @@ module.exports = function(context) {
             return;
         }
 
-        (node.raw.match(/\\[^\d]/g) || []).forEach(validate, {
-            "escapes": nodeEscapes,
-            "node": node
-        });
+        while ((match = pattern.exec(node.raw))) {
```

Comment:
Oh, that context object. Yes, I agree that could be more efficient, still
fairly readable, and worth doing.
On Apr 7, 2016 12:37 PM, "Burak Yiğit Kaya" <notifications@github.com>
wrote:

> In lib/rules/no-useless-escape.js
> <https://github.com/eslint/eslint/pull/5804#discussion_r58912906>:
>
> > @@ -95,10 +114,12 @@ module.exports = function(context) {
> >              return;
> >          }
> >
> > -        (node.raw.match(/\\[^\d]/g) || []).forEach(validate, {
> > -            "escapes": nodeEscapes,
> > -            "node": node
> > -        });
> > +        while ((match = pattern.exec(node.raw))) {
>
> @platinumazure <https://github.com/platinumazure> this one:
>
> {
>     "node": node
>     "escapes": nodeEscapes,
> }
>
> I actually think the following would be better and more efficient:
>
> var validator = validate.bind({ escapes, nodeEscapes, node: node });
> while ((match = pattern.exec(node.raw))) { validator(match); }
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/eslint/eslint/pull/5804/files/9b4a9f54c298cee99d563d2a43989353f8331c16#r58912906>
>